src/main.rs

```diff
@@ -0,0 +1,107 @@
+use miden_crypto::{
+    hash::rpo::RpoDigest,
+    merkle::{MerkleError, NodeIndex},
+    Felt, Word, ONE,
+    {hash::rpo::Rpo256, merkle::TieredSmt},
+};
+use std::time::Instant;
+
+fn main() {
+    construction(1000).unwrap();
+    construction(1_000_000).unwrap();
+
+    insertion(1000).unwrap();
+    insertion(1_000_000).unwrap();
+
+    proof_generation(1000).unwrap();
+    proof_generation(1_000_000).unwrap();
+}
+
+pub fn construction(size: u64) -> Result<(), MerkleError> {
+    let mut leaves = Vec::new();
+    for i in 0..size {
+        let key = Rpo256::hash(&rand_utils::rand_value::<u64>().to_be_bytes());
```

Comment:
Might be better to implement [Randomizable](https://github.com/facebook/winterfell/blob/main/utils/core/src/lib.rs#L239) on `RpoDigest` and generate random digests directly.